### PR TITLE
fix: Clean up and accelerate jest process + coverage integration

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

fix: clean up and accelerate jest process
fix: vscode integration
feat: add coverage generation (with vscode integration)

take a look to https://app.codecov.io/github/AmadeusITGroup/otter/tree/feature%2Fjest-cleanup to see CodeCov output

## Related issues

*- No issue associated -*

### CODE REVIEW

File: nx.json [hunk 1]
```diff
@@ -112,10 +112,13 @@
     "specs": [
       "base",
       "{projectRoot}/**/*.spec.ts",
+      "!{projectRoot}/**/*.it.spec.ts",
```
Comment (fpaul-1A) [inline nx.json:115]: this will also exclude the input from `test-int` that uses this named input

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  layout: "condensed_header, condensed_files, condensed_footer"
+  behavior: default
+
+component_management:
+  default_rules:
+    statuses:
+      - type: project
+        target: auto
+        branches:
+          - "!main"
```
Comment (mrednic-1A) [inline codecov.yml:14]: Is there a reason to deactivate it for `main` ?

**Review by fpaul-1A — APPROVED** [review]

**Review by fpaul-1A — DISMISSED** [review]

**Review by fpaul-1A — DISMISSED** [review]

**Review by fpaul-1A — APPROVED** [review]

File: .vscode/settings.json [hunk 3]
```diff
@@ -28,7 +28,10 @@
   "typescript.preferences.preferTypeOnlyAutoImports": true,
   "jest.jestCommandLine": "yarn jest",
   "jest.debugMode": true,
-  "jest.runMode": "on-save",
+  "jest.runMode": {
+    "type": "on-save",
```
Comment (matthieu-crouzet) [inline .vscode/settings.json:32]: I would prefer 'on-demand' (not blocking the PR for that)

File: jest.config.js [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-/** @type {import('ts-jest/dist/types').JestConfigWithTsJest} */
+/**
+ * Workspace global jest configuration
+ * Note: It is for IDE Test run and debugging purpose, you may want to run `yarn test` instead
+ * @type {import('ts-jest/dist/types').JestConfigWithTsJest}
+ **/
 module.exports = {
+  ...getJestGlobalConfig(__dirname),
+  coverageReporters: undefined,
+  reporters: undefined,
```
Comment (matthieu-crouzet) [inline jest.config.js:12]: Why do we remove the reporters ?

**Pull request merged**